client/src/components/all-components/SiteAccessibilityButton.js

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { actions as appActions } from "../../reducers/lab1/AppReducer";
+import { actions as mainActions } from "../../reducers/MainReducer";
+import { Panel as ColorPickerPanel } from "rc-color-picker";
+import AccessibilityImage from "../../assets/images/accessibility_icon.png";
+import { Collapse, Card, CardHeader, CardBody } from "reactstrap";
+import {
+  changeTSize,
+  setTextColor,
+  setBackgroundColor,
+} from "../footer/edit/editPage";
+
+const mapStateToProps = (state) => {
+  return {
+    // General
+    state: state,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    actions: bindActionCreators({ ...appActions, ...mainActions }, dispatch),
+  };
+};
+
+const SiteAccessibilityButton = () => {
+  const [open, setOpen] = useState(false);
+  const toggleCollapse = (e) => {
+    e.preventDefault();
+    if (e.target === e.currentTarget) {
+      setOpen(!open);
+    }
+  };
+
+  let state = {
+    fontSize: 0,
+    textColor: false,
+    bgColor: false,
+    displayColorPalette: false,
+    backgroundColor: null,
+    color: null,
+  };
+
+  function changeSize(size) {
+    const state_size = state.fontSize;
+    changeTSize(size);
+    state = { ...state, fontSize: state_size + size };
+  }
+
+  function onTextColorChange(obj) {
+    setTextColor(obj.color);
+    state = { ...state, color: obj.color };
+  }
+
+  function onBgColorChange(obj) {
+    setBackgroundColor(obj.color);
+    state = { ...state, backgroundColor: obj.color };
+  }
+
+  return (
+    <button
```

Comment:
The modal that pops is a child to the button. Is this what we want? Can we make it so that the modal is not a child? This seems a bit weird because hovering anywhere on the modal gives the `pointer-cursor`.

```
<>
    <Modal> ... </Modal>
    <Button> ... </Button>
<>
```

![image](https://github.com/user-attachments/assets/61ac2b89-e845-4152-a46d-7317182a59a5)

Reply:
fixed!